### PR TITLE
Fix #6248

### LINKED ISSUE #6248: :thumb_large size should only resize width to 300px
This is related to @Flaburgan 's comment [here](https://github.com/diaspora/diaspora/pull/6244#issuecomment-124727525):

> So, here is the real problem: thumb_large is not resized to be 300px width. It is its larger dimension, either height or width, which is resized to 300px. So, for images which have a big height and a small width, the result is catastrophic. Here is a good example: https://diaspora-fr.org/posts/901677

Furthermore, I think networks today are fast enough to support a little increase of that size. I think `thumb_large` size can be insreased to 400 or 500px.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/24909741-thumb_large-size-should-only-resize-width-to-300px?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
Change `thumb_large` to resize width to 300px and crop height to 500px (or other value) if necessary (would solve #6248),

### CODE REVIEW

**Comment by svbergerem** [comment]
`thumb_large` had a max-height and max-width of 300px, `scaled_full` has a max-width of 700px and no max-height which means that this PR will increase the traffic for mobile devices and we need to find a good tradeoff between image quality and filesize. (I won't propose any specific values.)

**Comment by AugierLe42e** [comment]
I don't think this is really important. Not more than hure GIF files that regularly appears on my stream.

**Comment by SuperTux88** [comment]
```thumb_large``` makes sense on the mobile interface to save traffic. Also the 300px preview is totally enough for the small phone screen. You can click on the image to see the big version.

**Comment by AugierLe42e** [comment]
> Also the 300px preview is totally enough for the small phone screen.

But not on large screens. On 1920px, it's repugnant. And I really don't think it is worth saving so few bandwidth.

**Comment by jhass** [comment]
You have a phone with 1920px of resolution?

**Comment by svbergerem** [comment]
> On 1920px, it's repugnant.

Oh, and I always thought we would optimize the mobile view for devices with small screens and limited internet access.

**Comment by AugierLe42e** [comment]
> You have a phone with 1920px of resolution?

Yes, but even on 768px, take a look:

ugly

And even on 500px screen size, it's still a bit blury:

capture d ecran 2015-07-24 a 20 18 38

**Comment by jhass** [comment]
Maybe we should just find a way to not scale it up but display it smaller instead?

**Comment by AugierLe42e** [comment]
I don't think it'll be elegant. But maybe we can use metadata about the browser (screen size, here) to display the right size ?

**Comment by SuperTux88** [comment]
Yes, I have a phone with 1920px, but no, I don't want a bigger version than 300px on the mobile-interface. 300px is ok for a 5" screen as preview. And you can click on it to see the bigger version.

If you use a tablet we should make the desktop-interface more usable for tablets, and you probably don't have traffic problems on a tablet. But the mobile-interface is optimized for phones and limited traffic and should remain so.

**Comment by AugierLe42e** [comment]
>  I don't want a bigger version than 300px on the mobile-interface. 300px is ok for a 5" screen as preview. 

But I do want and I don't think it's ok.

> If you use a tablet we should make the desktop-interface more usable for tablets, and you probably don't have traffic problems on a tablet. But the mobile-interface is optimized for phones and limited traffic and should remain so.

I don't have any traffic problems on mobile phone. Even with bigger size of images.

**Comment by AugierLe42e** [comment]
BTW, @SuperTux88, if you are really concerned about traffic usage, consider the following:

I am on stream page and I see a photo that is interesting but the resolution is so shitty that I'm obliged to display it full-size. So I click on it. I come to the SPV, but the resolution is still shitty, because it's the same. Now, I'm obliged to click again on the pic to display it full-size.

I have seen the picture and I'm happy, but now, I'm obliged to get back to the stream page and **entirely reload it** to see the rest of the stream.

In the move, I have fully loaded a useless page, the full photo and *2 times* the stream page.

Do you really think that when people are on suburbs and have unstable connection, they want to be obliged to go through this to just *correctly see an image* ?

**Comment by SuperTux88** [comment]
You can click directly on the image, no need to load the SPV first. And browser-back should use the browser-cache and not reload the page. And if I have less traffic or bad signal, I can decide to only view the 300px preview.

**Comment by AugierLe42e** [comment]
> And browser-back should use the browser-cache and not reload the page.

*Should*. It doesn't everywhere. 

>  I can decide to only view the 300px preview.

And what if *I* don't?

**Comment by Flaburgan** [comment]
@AugierLe42e I'm not with you on this one. We definitely need to load the small size version on mobile. If we don't do that here on the stream, then that means it's completely useless to generate small images. But I agree there is a problem with blur images in the mobile stream. I don't think it's about the **size** of the image loaded, I think it comes from the way we resize it, either during the generation of the image or because of the css classes we put on it. Your example here is not what I think about, for me the way it is displayed is fine. I'll try to find an example of what I'm talking about.

**Comment by Flaburgan** [comment]
So, here is the problem: thumb_large is not resized to be 300px width. It is its larger dimension, either height or width, which is resized to 300px. So, for images which have a big height and a small width, the result is catastrophic. Here is a good example: https://diaspora-fr.org/posts/901677

**Comment by AugierLe42e** [comment]
Wow... Nice catch ! Ok, closing it, then.

**Comment by AugierLe42e** [comment]
Still I think 300px is not enough for images.

**Comment by AugierLe42e** [comment]
Sorry, I'm reopening this because I'm actually not convinced with @Flaburgan explanation. Here is a clear example that illustrate it. The image is naturally landscape so does not enter into that bug. Here is how it appears on my phone:

screenshot_2015-07-25-00-44-05

And here is how it actually looks like when large:

screenshot_2015-07-25-00-44-22

I didn't touched the images, enlarged or shrank them. This really look like that on my phone. 

This is really unacceptable to me. It really gives impression of amateurism. Visual is more important than speed IMHO. Furthermore, HD and Full HD devices tend to be the norm now. They should be pivilieged as low resolution is a big flaw for them when high resolution is not for SD mobile.

**Comment by denschub** [comment]
So go ahead and fix the issue that causes the white boxes to be too big. 

> Visual is more important than speed IMHO.

Fine. Go ahead and tell that to the users using diaspora on a mobile network without a flat traffic rate.

**Comment by AugierLe42e** [comment]
> So go ahead and fix the issue that causes the white boxes to be too big. 

Sorry ? Which white boxes ?

**Comment by Flaburgan** [comment]
@AugierLe42e I still think there is a bug here, look how the white card goes all the way down on the first screenshot (this is what Dennis is talking about).

And about the resolution itself, we need number to take a decision.

**Comment by AugierLe42e** [comment]
> @AugierLe42e I still think there is a bug here, look how the white card goes all the way down on the first screenshot (this is what Dennis is talking about).

It's not a bug. There was content in this area that I anonymized.

**Comment by denschub** [comment]
Ah, that confused me. So everything is working well. 

> Can you please just stop always being ironic with me like that

That was neither irony nor especially directed to you. You are suggesting to increase the image size by a large factor, thus increasing the image size as well. This is the *mobile* interface which is used on *mobile* devices, usually using *mobile* network. There are a lot of people who have to pay for each megabyte transfered. This is why the image size is limited to a very small format, and users can view the full size image if they want to. They are not forced to load tons of large images in the stream.

If you think I am wrong (which you obviously do), head over to Loomio and start a proposal.

**Comment by AugierLe42e** [comment]
Ok, [here](https://www.loomio.org/d/g0Ni9uSR/change-the-image-size-in-the-mobile-view#comment-739874) is the proposition.

**Comment by AugierLe42e** [comment]
So, if the loomio proposition is accepted, here is what I'm planing to do:

* add a parameter to let the user display hi-res images on stream, unchecked by default,
* change `thumb_large` to resize width to 300px and crop height to 500px (or other value) if necessary (would solve #6248),
* add a `thumb_huge` size with width resized to 700px and height croped to 900px (or other value) if necessary,
* add `max-height: 900px;` to the photo area so that images with big height do not oblige the user to scroll too much in stream.

I would basically work like this:

* if hi-res option is unchecked, nothing changes from now,
* if hi-res option is checked, `thumb_huge` is displayed if exists, else, `scaled_full`.

Is any of this controversial and/or impossible?

**Comment by denschub** [comment]
Apparently, nobody has objections. I'm looking forward to your updated PR.

**Comment by goobertron** [comment]
Pixel dimensions isn't the only consideration. Even if a phone's screen is 800px wide, if it's only 3" across, the photo will still be small to the eye, and so a smaller image resolution would still be fine. I have *never* seen a phone the size of those screenshots above!

**Comment by AugierLe42e** [comment]
> Pixel dimensions isn't the only consideration. Even if a phone's screen is 800px wide, if it's only 3" across, the photo will still be small to the eye, and so a smaller image resolution would still be fine. I have never seen a phone the size of those screenshots above!

OnePlus One.

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+          format.js{ render json: {
+                                    photo_id: @photo.id,
+                                    image_url: @photo.url(:thumb_large),
+                                    image_url_huge: @photo.url(:thumb_huge),
+                                    image_url_medium: @photo.url(:thumb_medium),
+                                    image_url_small: @photo.url(:thumb_small),
+                                    author_id: author_id
+                                  },
+                            status: 201
+                   }
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: `end` at 84, 19 is not aligned with `format.js{ render json: {` at 75, 10

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
     @photo = Photo.where(:id => params[:photo_id], :author_id => author_id).first
 
     if @photo
-      profile_hash = {:image_url        => @photo.url(:thumb_large),
-                      :image_url_medium => @photo.url(:thumb_medium),
-                      :image_url_small  => @photo.url(:thumb_small)}
+      profile_hash = {image_url: @photo.url(:thumb_large),
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 3]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                      image_url_medium: @photo.url(:thumb_medium),
+                      image_url_huge: @photo.url(:thumb_huge),
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 4]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                      image_url_small: @photo.url(:thumb_small)}
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 5]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
 
       if current_user.update_profile(profile_hash)
         respond_to do |format|
-          format.js{ render :json => { :photo_id  => @photo.id,
-                                       :image_url => @photo.url(:thumb_large),
-                                       :image_url_medium => @photo.url(:thumb_medium),
-                                       :image_url_small  => @photo.url(:thumb_small),
-                                       :author_id => author_id},
-                            :status => 201}
+          format.js{ render json: {
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 6]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                    photo_id: @photo.id,
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 7]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                    image_url: @photo.url(:thumb_large),
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 8]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                    image_url_huge: @photo.url(:thumb_huge),
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 9]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                    image_url_medium: @photo.url(:thumb_medium),
+                                    image_url_small: @photo.url(:thumb_small),
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 10]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                    author_id: author_id
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb (hunk 6 above)
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.

File: app/controllers/photos_controller.rb [hunk 11]
```diff
@@ -65,18 +65,23 @@ def make_profile_photo
...
+                                  },
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Indent the right brace the same as the start of the line where the left brace is.

File: app/controllers/photos_controller.rb (hunk 5 above)
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Block body expression is on the same line as the block start.

Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Space missing to the left of {.

File: app/uploaders/processed_image.rb [hunk 12]
```diff
@@ -17,17 +17,30 @@ def filename
...
+      _cols, rows = img[:dimensions]
+      top_to_crop = [((rows - height) / 2), 0].max
+      bottom_to_crop = [(rows - height - top_to_crop), 0].max
+      img.crop "#{width}x#{height}+#{top_to_crop}+#{bottom_to_crop}"
+    end
+  end
+
   version :thumb_small do
-    process :resize_to_fill => [50,50]
+    process resize_to_fill: [50,50]
```
Comment (diaspora-code-review) [inline app/uploaders/processed_image.rb]: Space missing after comma.

File: app/uploaders/processed_image.rb [hunk 13]
```diff
@@ -17,17 +17,30 @@ def filename
...
   end
   version :thumb_medium do
-    process :resize_to_limit => [100,100]
+    process resize_to_limit: [100,100]
```
Comment (diaspora-code-review) [inline app/uploaders/processed_image.rb]: Space missing after comma.

File: app/uploaders/processed_image.rb [hunk 14]
```diff
@@ -17,17 +17,30 @@ def filename
...
   version :thumb_large do
-    process :resize_to_limit => [300,300]
+    process resize_and_crop: [300, 500]
+  end
+  version :thumb_huge do
+    process resize_and_crop: [700, 900]
   end
   version :scaled_full do
-    process :resize_to_limit => [700,nil]
+    process resize_to_limit: [700,nil]
```
Comment (diaspora-code-review) [inline app/uploaders/processed_image.rb]: Space missing after comma.

File: app/controllers/photos_controller.rb [hunk 15]
```diff
@@ -62,54 +62,60 @@ def create
...
       end
     else
-      respond_with photo, :location => person_photos_path(current_user.person)
+      respond_with photo, location: person_photos_path(current_user.person)
     end
   end
 
   def edit
-    if @photo = current_user.photos.where(:id => params[:id]).first
+    if @photo = current_user.photos.where(id: params[:id]).first
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Assignment in condition - you probably meant to use `==`.

**Comment by AugierLe42e** [comment]
This commit add point 2 and 3.

File: app/controllers/photos_controller.rb [hunk 16]
```diff
@@ -63,27 +63,27 @@ def create
...
 
       if current_user.update_profile(profile_hash)
         respond_to do |format|
-          format.js{ render :json => { :photo_id  => @photo.id,
-                                       :image_url => @photo.url(:thumb_large),
-                                       :image_url_medium => @photo.url(:thumb_medium),
-                                       :image_url_small  => @photo.url(:thumb_small),
-                                       :author_id => author_id},
-                            :status => 201}
+          format.js{ render json: { photo_id:         @photo.id,
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Space missing to the left of {.

Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Block body expression is on the same line as the block start.

Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Space inside { detected.

File: app/controllers/photos_controller.rb [hunk 17]
```diff
@@ -63,27 +63,27 @@ def create
...
+                                    image_url:        @photo.url(:thumb_large),
+                                    image_url_medium: @photo.url(:thumb_medium),
+                                    image_url_small:  @photo.url(:thumb_small),
+                                    author_id: author_id},
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/controllers/photos_controller.rb [hunk 18]
```diff
@@ -63,27 +63,27 @@ def create
...
+                            status: 201}
```
Comment (diaspora-code-review) [inline app/controllers/photos_controller.rb]: Expression at 80, 40 should be on its own line.

File: app/uploaders/processed_image.rb [hunk 19]
```diff
@@ -17,17 +17,23 @@ def filename
     model.random_string + File.extname(@filename) if @filename
   end
 
+  def resize_and_crop(width, height)
+    resize_to_limit(width, nil) do |img|
+      img.crop "#{width}x#{height}>+#{0}+#{100}"
```
Comment (diaspora-code-review) [inline app/uploaders/processed_image.rb]: Literal interpolation detected.

File: app/uploaders/processed_image.rb [hunk 20]
```diff
@@ -17,17 +17,23 @@ def filename
...
+      img.crop "#{width}x#{height}>+#{0}-#{100}"
```
Comment (diaspora-code-review) [inline app/uploaders/processed_image.rb]: Literal interpolation detected.

**Comment by AugierLe42e** [comment]
I've changed my mind. It'll be easier to just solve #6248 and add the Blueimp gallery on mobile view too.

**Comment by AugierLe42e** [comment]
@jhass: Can you give me a feedback on this code?

**Comment by jhass** [comment]
I don't feel like I can estimate whether this will result in better or worse results for the majority of uploads. I guess somebody would have to try it out for a while.

**Comment by AugierLe42e** [comment]
No, I meant, from a technical point of view.
@Flaburgan: do you feel like merging this into d-fr and test it for a while?

**Comment by jhass** [comment]
There's not much to review there...

**Comment by AugierLe42e** [comment]
At least, now I'm sure ;)

**Comment by AugierLe42e** [comment]
@Flaburgan: ping.

**Comment by Flaburgan** [comment]
Hi!

As I hate take decision based on "I have the feeling that..." and I prefer facts and numbers, I did the test to see if it's worth cropping or not.

Here are some images picked on diaspora*. A normal picture and then graphical info because those are the type of images which have a big height.

| image | raw | resized 300px width | resized 300px width and cropped 500px height |
|---------|------|------|------|
| C3PO | [992ko](https://cloud.githubusercontent.com/assets/930064/11743006/8e3ae8f4-a004-11e5-8c15-aebdaeceef84.png) | [62ko](https://cloud.githubusercontent.com/assets/930064/11743005/8e38c358-a004-11e5-9704-7bcefee4a26f.jpg) | not cropped | 
| Info 1 | [957ko](https://cloud.githubusercontent.com/assets/930064/11743001/8e218b16-a004-11e5-8665-73dc98d91fe8.jpeg) | [215ko](https://cloud.githubusercontent.com/assets/930064/11743000/8e1ef50e-a004-11e5-96d6-2073bdaa64c3.jpeg) | [62ko](https://cloud.githubusercontent.com/assets/930064/11743003/8e2962dc-a004-11e5-8f39-e768e43da640.jpeg) |
| Info 2 | [1.8mo](https://cloud.githubusercontent.com/assets/930064/11743007/8e3f647e-a004-11e5-9b41-74e7b6503bce.jpg) | [309ko](https://cloud.githubusercontent.com/assets/930064/11743008/8e40f37a-a004-11e5-964c-b444fab0b209.jpg) | [107ko](https://cloud.githubusercontent.com/assets/930064/11743009/8e41e550-a004-11e5-9507-e6825dd51b59.jpg) |
| Info 3 | [143ko](https://cloud.githubusercontent.com/assets/930064/11743119/384e9e4e-a005-11e5-82c2-a6a420d8f3a5.jpg) | [48ko](https://cloud.githubusercontent.com/assets/930064/11743002/8e26c982-a004-11e5-946f-6c05ee8b6660.jpg) | [40ko](https://cloud.githubusercontent.com/assets/930064/11743004/8e2d37cc-a004-11e5-9edf-b051cb9b463b.jpg)

So, even big images like info 2 which has a raw height of 5000px for a width of 1000px and is 1.8Mo goes down to 300ko only by resizing to 300px width. Is that enough or do we want something lighter?

In any case, the current implementation is not good in my opinion because nothing tells the user he is not watching the full image and is missing a part of it and this is clearly bad UX. I guess if we want to do that we need something like a "show more" link.

**Comment by AugierLe42e** [comment]
> So, even big images like info 2 which has a raw height of 5000px for a width of 1000px and is 1.8Mo goes down to 300ko only by resizing to 300px width. Is that enough or do we want something lighter?

I think the good point is more a design one. The stream will be easier to read displaying images with limited height.

**Comment by Flaburgan** [comment]
>The stream will be easier to read displaying images with limited height.

This is the role of the "show more" feature.

**Comment by AugierLe42e** [comment]
> This is the role of the "show more" feature.

Note that I also plan to add the gallery to the mobile view too. This will easier images disaplying.

**Comment by AugierLe42e** [comment]
Hey, this PR has been opened for a year now. What do we decide about it? Should I close it or can it be merged?

**Comment by Flaburgan** [comment]
As said, I'm not in favour of a crop without any indication to the user that he is not viewing the full image. At the moment, I would simply change `process :resize_to_limit => [300,300]` to `process :resize_to_limit => [300,nil]`.

**Comment by denschub** [comment]
> At the moment, I would simply change `process :resize_to_limit => [300,300]` to `process :resize_to_limit => [300,nil]`.

I wouldn't. Doing so would allow 300x30000px images to be rendered with their full filesize. And yes, simply cropping is bad, too. How about, let's say, `[300,1500]`, just to have a safety net if people go crazy. ;)

**Comment by AugierLe42e** [comment]
Why would simply cropping be bad?

**Comment by Flaburgan** [comment]
@AugierLe42e as said just above, because nothing in the interface indicates to the user that he's not seeing the full image so he can't know he has to click on it to view the real content.

**Comment by AugierLe42e** [comment]
Maybe we can add a "See full image" in a corner? I like the idea of cropping because it limits the size of the displayed image in the stream while keeping a correct quality.

**Comment by AugierLe42e** [comment]
Something like that, maybe?

stream

**Comment by denschub** [comment]
No.

* Localizing this text would be impossible if it's rendered within the image since we don't know what the language will be when it's embedded.
* Preprocessing thumbnails even further takes more time, and in fact, this is one of the most expensive workers so far.
* People don't want us to alter their images in any way. Cropping may remove essential parts of the images if they are off-center, which most art-y photos for example are.
* If you crop something, the user still would have to click on the image to see the "usable" version. You're entire argument of improving the UX you've set up in earlier comments is worthless here.

Both Fla and I have made reasonable limits for sizing the image down, while keeping a higher resolution *and* the full contents. There is common consent from everyone else that cropping is the wrong idea here, so I don't really know what your intentions are.

**Comment by AugierLe42e** [comment]
Wow, easy! I was just proposing a solution. I think that cropping is a better solution, that's it, but if nobody wants it, then, ok. It's no big deal. No need to be aggressive :/

**Comment by goobertron** [comment]
Augier, do try not to take criticism so personally. I see no aggression from Dennis, simply an attempt to explain really clearly to you the reasons not to use automatic image cropping, because you had not listened to the previous comments saying no to cropping. I was going to say something similar to Dennis about the issues with processor resources (although without so much technical knowledge) and the artistic problems of automatic cropping, but he got there first.

One problem with automatic image cropping which Dennis raises is that of images in which the focal point is not in the centre, which is common. How would your software choose where best to crop this image, for instance?

foxcone

**Comment by AugierLe42e** [comment]
This image wouldn't be cropped: it's landscape :)

**Comment by goobertron** [comment]
Oh yeh, I meant to take a vertical photo! But you take the point.

**Comment by SuperTux88** [comment]
Superseded by #7383.

**Pull request closed without merging**